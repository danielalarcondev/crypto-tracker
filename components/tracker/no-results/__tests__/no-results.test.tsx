import * as React from 'react';
import '@testing-library/jest-dom/jest-globals';

import { fireEvent, render, screen } from '@testing-library/react';
import { expect, it, describe } from '@jest/globals';
import NoResults from '@components/tracker/no-results/no-results';

describe('Assets No Results: ', () => {
	
    it('should render correctly', () => {
        const { container } = render(<NoResults onRetry={jest.fn()} />);

        const message = screen.getByTestId('assents-no-results-message');
		
        expect(message?.textContent).toBe('No results. Would you like to try again?');
        expect(container).toMatchSnapshot();
    });
	
    it('should call onRetry callback on retry button click', () => {
        const onRetry = jest.fn();
        render(<NoResults onRetry={onRetry} />);

        const button = screen.getByTestId('assets-no-results-retry-button');
        fireEvent.click(button);
		
        expect(onRetry).toHaveBeenCalled();
    });
});
