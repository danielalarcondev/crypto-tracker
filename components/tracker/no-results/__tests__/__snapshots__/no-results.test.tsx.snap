// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assets No Results:  should render correctly 1`] = `
<div>
  <div
    class="w-[26rem] h-32 bg-red-300 rounded-md p-6"
    data-testid="assents-no-results"
  >
    <h1
      class="text-md mb-4"
      data-testid="assents-no-results-message"
    >
      No results. Would you like to try again?
    </h1>
    <button
      class="group relative flex items-stretch justify-center p-0.5 text-center font-medium transition-[color,background-color,border-color,text-decoration-color,fill,stroke,box-shadow] focus:z-10 focus:outline-none bg-blue-500/80 hover:bg-blue-500 border border-white text-white rounded-lg w-full"
      data-testid="assets-no-results-retry-button"
      type="button"
    >
      <span
        class="flex items-stretch transition-all duration-200 rounded-md px-4 py-2 text-sm"
      >
        Retry
        <svg
          class="w-4 h-4 ml-2 flex self-center"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.023 9.348h4.992v-.001M2.985 19.644v-4.992m0 0h4.992m-4.993 0 3.181 3.183a8.25 8.25 0 0 0 13.803-3.7M4.031 9.865a8.25 8.25 0 0 1 13.803-3.7l3.181 3.182m0-4.991v4.99"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
